tests/net/lib/prometheus/counter/src/main.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+#include <zephyr/net/prometheus/counter.h>
+
+struct prometheus_metric test_counter_metric = {
+	.type = PROMETHEUS_COUNTER,
+	.name = "test_counter",
+	.description = "Test counter",
+	.num_labels = 1,
+	.labels = {{
+		.key = "test",
+		.value = "counter",
+	}},
+};
+
+PROMETHEUS_COUNTER_DEFINE(test_counter_m, &test_counter_metric);
+
+/**
+ * @brief Test prometheus_counter_inc
+ * @details The test shall increment the counter value by 1 and check if the
+ * value is incremented correctly.
+ */
+ZTEST(test_counter, test_prometheus_counter_inc)
+{
+	int ret;
+
+	zassert_true(test_counter_m.value == 0, "Counter value is not 0");
+
+	ret = prometheus_counter_inc(&test_counter_m);
+	zassert_true(ret == 0, "Error incrementing counter");
```

Comment:
I recently learned about [`zassert_ok(ret, "msg");`](https://docs.zephyrproject.org/latest/doxygen/html/group__ztest__assert.html#gade40e2ec78ec813739e7725524fae7f0).
Same observation applies to other tests.

Reply:
Me too i recently learned and applied to your suggestions each test cases :)